### ISSUE TITLE
AutocompleteArrayInput: duplicate suggestions

### ISSUE THREAD

**Issue by LoicMahieu** [marmelab/react-admin#3826]
**What you were expecting:**

Don't have any duplicate suggestions in the `AutocompleteArrayInput`.

**What happened instead:**

Using the `ReferenceArrayInput`:

```jsx
<ReferenceArrayInput
  source="relatedPosts"
  label="Related posts"
  reference="posts"
>
  <AutocompleteArrayInput optionText="title" fullWidth />
</ReferenceArrayInput>
```

When editing a record that already have some `relatedPosts`, when you delete some of them, it produces duplicated suggestions. See the video:
![Kapture 2019-10-16 at 10 09 42](https://user-images.githubusercontent.com/900947/66900490-30713280-effd-11e9-870d-9039c29faa66.gif)

**Steps to reproduce:**

- Go to sandbox https://codesandbox.io/s/simple-p40b6 or view https://p40b6.csb.app/#/posts
- Edit a record, add some "related posts" and then save the record
- Go back to edit and remove some related posts.
- You'll see duplicated suggestions and some React warnings with ` Encountered two children with the same key` 

**Related code:**

* CodeSandbox: https://codesandbox.io/s/simple-p40b6

**Other information:**

**Environment**

* React-admin version: 3.0.0-beta
* Last version that did not exhibit the issue (if applicable):
* React version: 16.9
* Browser: Chrome
* Stack trace (in case of a JS error):

**Comment by LoicMahieu** [comment]
It seems to be related to the `idsToFetch` that are set in the [`useReferenceArrayInputController.ts#L54-L61`](https://github.com/marmelab/react-admin/blob/v3.0.0-beta.0/packages/ra-core/src/controller/input/useReferenceArrayInputController.ts#L54-L61).

I tried to remove the `newIdsToFetch.length > 0` and it seems to work well but I know if it's a good change:
```diff
diff --git a/packages/ra-core/src/controller/input/useReferenceArrayInputController.ts b/packages/ra-core/src/controller/input/useReferenceArrayInputController.ts
index 8aa027fa..0d9007ff 100644
--- a/packages/ra-core/src/controller/input/useReferenceArrayInputController.ts
+++ b/packages/ra-core/src/controller/input/useReferenceArrayInputController.ts
@@ -54,9 +54,7 @@ const useReferenceArrayInputController = ({
     useEffect(() => {
         const newIdsToFetch = difference(input.value, inputValue.current);

-        if (newIdsToFetch.length > 0) {
-            setIdsToFetch(newIdsToFetch);
-        }
+          setIdsToFetch(newIdsToFetch);
         inputValue.current = input.value;
     }, [input.value, setIdsToFetch]);
```

**Comment by LoicMahieu** [comment]
I also encountered a another issue but I don't think it is related:
![Kapture 2019-10-16 at 14 34 07](https://user-images.githubusercontent.com/900947/66919772-47764b80-f022-11e9-8e0c-5556f418fd09.gif)

Steps:
- From empty value, add 2 items.
- Focus out
- Focus and tap any chars that doesn't match with any item
- See: some of chip doesn't have text.

**Comment by fzaninotto** [comment]
Fixed by #3898
